src/gc/gc.cpp

```diff
@@ -35739,7 +35745,13 @@ GCHeap::SetCardsAfterBulkCopy( Object **StartPoint, size_t len )
     end = StartPoint + (len/sizeof(Object*));
     while (rover < end)
     {
-        if ( (((uint8_t*)*rover) >= g_gc_ephemeral_low) && (((uint8_t*)*rover) < g_gc_ephemeral_high) )
+        // For Workstation GC, where there is exactly one ephemeral region, we can avoid touching the
+        // card table if the rover is currently pointing into a region outside of the ephemeral generation.
+        //
+        // For Server GC, where there are multiple ephemeral regions, we must always touch the card table.
+#ifndef MULTIPLE_HEAPS
+        if ( (((uint8_t*)*rover) >= gc_heap::ephemeral_low) && (((uint8_t*)*rover) < gc_heap::ephemeral_high) )
```

Comment:
Agree - it may be best to move this to the EE side, and remove it from the GC/EE interface. It is on the EE side in CoreRT already: https://github.com/dotnet/corert/search?utf8=%E2%9C%93&q=InlinedBulkWriteBarrier